examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -464,9 +468,26 @@ get_thread_state(struct pt_regs *ctx) {
 
 found:
   // Get pointer to top frame from PyThreadState
-  bpf_probe_read_user(
+  if (state->offsets.PyThreadState.frame > -1) {
+    // For Python <= 3.10 get frame pointer directly from PyThreadState
```

Comment:
3.10 or 3.11?

Reply:
TIll 3.10 (`<=3.10`) the pointer is right here, in ThreadState.